src/main/java/org/sonar/plugins/jacoco/JacocoSensor.java

```diff
@@ -51,20 +52,23 @@ public void execute(SensorContext context) {
   void importReports(ReportPathsProvider reportPathsProvider, FileLocator locator, ReportImporter importer) {
     Collection<Path> reportPaths = reportPathsProvider.getPaths();
     if (reportPaths.isEmpty()) {
-      LOG.debug("No reports found");
+      LOG.info("No report imported, no coverage information will be imported by JaCoCo XML Report Importer");
       return;
     }
 
+    String fileList = reportPaths.stream().sorted().limit(MAX_LOGGED_FILE_NAMES).map(Path::toString).collect(Collectors.joining(";"));
+    if (reportPaths.size() > MAX_LOGGED_FILE_NAMES) {
+      fileList += ";... ";
```

Comment:
I'm not at all convinced by this. I feel that a partial reporting is not that useful and will trigger work on the long term.

What if some files are missing their coverage and the file containing it is not in the list, or the file IS the list but located only after the 10 firsts. I would prefer to log the number of reports in clear, and maybe in debug the exhaustive full list of files. With an INFO log such as:

```
Importing 14 coverage report(s). Turn your logs in debug mode in order to see the exhaustive list.
```

Reply:
This is definitely a better alternative.